src/com2ann.py

```diff
@@ -832,65 +833,111 @@ def translate_file(infile: str, outfile: str, options: Options) -> None:
 
 
-def main() -> None:
+def parse_cli_args(args: Sequence[str] | None = None) -> dict[str, Any]:
     parser = argparse.ArgumentParser(description=__doc__)
-    parser.add_argument("-o", "--outfile",
-                        help="output file or directory, will be overwritten if exists,\n"
-                             "defaults to input file or directory")
     parser.add_argument("infile",
+                        nargs="*",
+                        type=pathlib.Path,
                         help="input file or directory for translation, must\n"
                              "contain no syntax errors;\n"
-                             "if --outfile is not given, translation is\n"
-                             "made *in place*")
+                             "if --outfile is not given or multiple input files are\n"
+                             "given, translation is made *in place*")
+
+    parser.add_argument("-o", "--outfile",
+                        type=pathlib.Path,
+                        help="output file or directory, will be overwritten if exists,\n"
+                             "defaults to input file or directory. Cannot be used\n"
+                             "when multiple input files are defined. infile and\n"
+                             "outfile must be of the same type (file vs. directory)")
+
     parser.add_argument("-s", "--silent",
                         help="do not print summary for line numbers of\n"
                              "translated and rejected comments",
                         action="store_true")
+
     parser.add_argument("-n", "--drop-none",
                         help="drop any None as assignment value during\n"
                         "translation if it is annotated by a type comment",
                         action="store_true")
+
     parser.add_argument("-e", "--drop-ellipsis",
                         help="drop any Ellipsis (...) as assignment value during\n"
                         "translation if it is annotated by a type comment",
                         action="store_true")
+
     parser.add_argument("-i", "--add-future-imports",
                         help="add 'from __future__ import annotations' to any file\n"
                         "where type comments were successfully translated",
                         action="store_true")
+
     parser.add_argument("-w", "--wrap-signatures",
                         help="wrap function headers that are longer than given length",
                         type=int, default=0)
+
     parser.add_argument("-v", "--python-minor-version",
                         help="Python 3 minor version to use to parse the files",
                         type=int, default=-1)
 
-    args = parser.parse_args()
-    if args.outfile is None:
-        args.outfile = args.infile
+    options = parser.parse_args(args)
+
+    infile = options.infile  # type: List[pathlib.Path]
+    outfile = options.outfile  # type: Optional[pathlib.Path]
```

Comment:
Please use the plural form so your reader knows that this is not just one single input file but is instead a list of files.
```suggestion
    infiles: list[pathlib.Path] = options.infiles
    outfile:  pathlib.Path | None = options.outfile
```